src/merkle/merkle_tree.rs

```diff
@@ -80,7 +80,7 @@ impl MerkleTree {
             return Err(MerkleError::InvalidIndex(depth, index));
         }
 
-        let pos = 2usize.pow(depth as u32) + (index as usize);
+        let pos = 2usize.pow(depth) + (index as usize);
```

Comment:
minor nit: the convention we use for specifying types is `value_type`. So, here it would be `2_usize`.

Reply:
Fixed!